blueprints/datamodel.tex

```diff
@@ -335,7 +335,117 @@ \subsection{Delivery Services}
+Delivery Services.
+
+\paragraph{Anonymous Blocking}
+A Delivery Service that has "Anonymous Blocking" tells Traffic Router to block
+requests from anonymized IP addresses. Whether or not and how well that can
+actually be done is dependent on the configuration of each Traffic Router itself,
+and if this Delivery Service is DNS-routed the only IP address Traffic Routers
+will be capable of checking for anonymization (e.g. known proxy/VPN/TOR exit node)
+then
+
+\paragraph{Deep Caching}
+
+\paragraph{Caching}
+\begin{itemize}
+	\item \code{CACHE}
+	\item \code{RAM\_ONLY}
+	\item \code{NO\_CACHE}
+\end{itemize}
+
+%TODO - STEERING doesn't have this?
+\paragraph{Cache Servers}
+A Delivery Service's assigned Cache Servers are expressed as a set of Cache Server
+IDs.
+
+\paragraph{CDN}
+
+%TODO: use-case for check path?
+
+\paragraph{Denied Access Redirect}
+
+\paragraph{DSCP}
+
+\paragraph{ECS}
+
+\paragraph{Geographic Limiting}
+This property describes limitations to the availability of this Delivery Service's
+content on the basis of the requesting client's geographic location. It is a set of
+strings, each of which is an \href{https://www.iso.org/obp/ui/#search/code/}{ISO 3166-1}
+alpha-2 country code, optionally with ISO 3166-2 subdivisional alphabetic code. This is
+a "white list" of countries/subdivisions wherein content is to be made available\footnote{
+This property is meant to inform Traffic Router; Cache Servers cannot be relied upon to
+approve or deny access on a geographic basis. Thus, if routing is bypassed, restricted
+content is totally accessible to requesting clients.}.\\
+Content is \emph{always} available to clients whose IP addresses are found within the
+Traffic Routers' Coverage Zone File(s). With that in mind, when this property is an empty
+set it means that no geographic regions are "whitelisted" and thus \emph{only} clients
+whose IP addresses are found within a Coverage Zone File will be granted access to content.
+When this property has a "Null" type, there is no geographic restriction placed on the
+Delivery Service's content access.
+
+\paragraph{Maximum Origin Connections}
+
+%TODO: use-case for header-rewrite text?
+
+\paragraph{Miss Location}
+
+\paragraph{Name}
+A Delivery Service's "Name" is a string that uniquely identifies it among all
+Delivery Services. It MUST only contain alphanumerics, hyphens, underscores and
+spaces, and MUST NOT begin with a non-alphanumeric character nor end with a
+non-alphanumeric character. This is used to generate part of the default request
+hostnames by replacing all non-alphanumeric characters with a hyphen.
+
+\paragraph{Notes}
+This section is an arbitrary string containing miscellaneous, human-friendly
+information about the Delivery Service. Other ATC components SHOULD NOT parse
+this for specific information fields, or expect it to be in a particular format.
+
+%TODO: MSO use-case?
+\paragraph{Origin}
+
+\paragraph{Query String Handling}
```

Comment:
I believe we should kill this option all together and force people to actually think about their cachekey.  Most HTTP security classes that mention CDNs talk specifically about this concept.

Reply:
Well how would that be implemented today using ATS configuration files? I was under the impression that cache key configuration required a plugin that was totally separate from remap.config

Reply:
We do it today using a combination of raw remap and DS profile/parameters.  This is required today for every base origin url field change or you end up breaking existing caches.

Reply:
Wait, so every time you change the Delivery Service's orgServerFqdn you have to change the cache key using raw remap.config lines? Why? And what Parameters?

Reply:
Correct.  If you don't manually assert the cachekey config before changing the base origin url field, you effectively invalidate the entire cache of that ds.